lib/sbi_ipi.c

```diff
@@ -70,6 +80,51 @@ void sbi_ipi_clear_smode(struct sbi_scratch *scratch)
 	csr_clear(CSR_MIP, MIP_SSIP);
 }
 
+static void sbi_ipi_tlb_flush_all(unsigned long asid)
+{
+	__asm__ __volatile__ ("sfence.vma x0, %0"
+				: : "r" (asid)
+				: "memory");
+}
+
+static void sbi_ipi_sfence_vma(struct sbi_tlb_info *tinfo)
+{
+	unsigned long start = tinfo->start;
+	unsigned long size = tinfo->size;
+	unsigned long i;
+
+	if (start == 0 && size == 0)
+		sbi_ipi_tlb_flush_all(0);
```

Comment:
This is wrong. Flush ASID 0 != global flush. All pages marked with G will now not be flushed at all. And why isn't there a check for SBI_TLB_FLUSH_ALL?

Reply:
Ahh okay. Thanks for pointing out. I misread the spec (got confused x0 with value 0).
I will fix it right now.